### ISSUE TITLE
make dist-bzip2 broken

### ISSUE THREAD

**Issue by JBBIntel** [dun/munge#45]
Having installed the latest commit from master, which fixed a problem with the specs file, I installed munge, with make && make install

Next, I need to make an rpm, for xCAT to use in provisioning. So I ran
make dist-bzip2 PACKAGE=munge VERSION=0.5.11

This produces a tar.bz2 file, but when I try to run rpmbuild against it;
 rpmbuild -tb --clean munge-0.5.11.tar.bz2
I get an error;
error: File /home/jbberry/slurm/munge/V0.5.11/munge-0.5.11.tar.xz: No such file or directory

**Comment by dun** [comment]
The "dist" makefile targets haven't been set up to work properly. I'm not sure how you're able to produce a tar.bz2 file above -- it shouldn't work. It should fail with:
```
make: *** No rule to make target 'DISCLAIMER', needed by 'distdir'.  Stop.
```

But from the output you provided, you're failing in `rpmbuild` because of the switch to `xz` in 8cd3e16e46402359a7c89693ee9418783e6189fb.

Regardless, `make dist` should be able to create a distribution archive for use with `rpmbuild`.

**Comment by JBBIntel** [comment]
Yeah, it did fail with that. So I fooled with the spec file a little, to see if it was just one glitch and I could get around it. But it was not, so I decided to contact you.

I am not following your answer. If
   The "dist" makefile targets haven't been set up to work properly
Then how can it be that
  Regardless, make dist should be able to create a distribution archive for use with rpmbuild.

As I recall, make dist ran without error, but I could not find an rpm file anywhere.

  Thanks,

   Jerome

**Comment by dun** [comment]
OK, let me rephrase.

The "dist" makefile targets haven't been set up to work properly. I don't see how they could work in their current state. It's reasonable to expect that you can create a tarball with `make dist` and feed that into `rpmbuild`. But that currently doesn't work (as you've noted) and needs to be fixed. It hasn't been an issue for me since I've always created my distribution archives in another manner.

Yet you somehow created a **tar.bz2** file. And based on your output, `rpmbuild` failed because it was looking for a **tar.xz** file due to the change of the `munge.spec` `Source0` tag in 8cd3e16e46402359a7c89693ee9418783e6189fb.

**Comment by JBBIntel** [comment]
The reason I was able to create a bz2 was because I removed that dependency (DISCLAIMER) from the spec file, hoping I was dealing with a single bug. No such luck.

I have started over, by downloading your latest commit, which fixes that problem. Once I got the dependencies squared away (libopenssl-devel), I was able to configure. But make fails;

f045:/home/jbberry/slurm/munge/V0.5.11/commit/munge-0.5.11 # make
make: *** No rule to make target 'META', needed by 'config.status'.  Stop.

Am I going about this wrong? I want to install munge under SLES12.0. Once I am sure it is working with slurm, I need to build an rpm, so I can install routinely on rtebooted machines using xCAT. Any guidance is welcome. Thanks for your help.

   Jerome

**Comment by dun** [comment]
It looks like you removed the `META` file which `make` doesn't know how to rebuild.

Until I get `make dist` working, you could build an rpm from `HEAD` via:
```
git clone --depth=1 https://github.com/dun/munge.git munge-0.5.11
rm -rf munge-0.5.11/.git*
tar cJf munge-0.5.11.tar.xz munge-0.5.11
rpmbuild -ta --clean munge-0.5.11.tar.xz
```

**Comment by JBBIntel** [comment]
Hello,

   That led to a bunch of issues with bzip2 and its dependencies. I finally resolved that, I think, and now I  am getting this error;

RPM build errors:
    Installed (but unpackaged) file(s) found:
   /usr/lib64/libmunge.a

After poking around on the web a bit, I gather that I can either add libmunge.a to the %files list in the munge.spec file, or there are a couple ways I can make rpmbuild work in spite of the error. Which would you recommend?

   Thanks again,

   Jerome

**Comment by dun** [comment]
Packaging guidelines recommend excluding static libraries:
https://fedoraproject.org/wiki/Packaging:Guidelines#Packaging_Static_Libraries

If you're using the `munge.spec` from 2b135195587efb1894a91fd7df254b3c47a0c962, `configure` is called with `--disable-static` to prevent building the static library (`libmunge.a`). It should not be showing up as an installed (but unpackaged) file.

If you're using the `munge.spec` before that, the static library was listed under `%files devel` via `%{_libdir}/*.a`.
